php/class-hubspot-contacts.php

```diff
@@ -360,12 +371,25 @@ protected function process_get_contact_response( $response ) {
 	 * @param string $email
 	 * @return boolean false if it does not successfully create a contact in the Hubspot API
 	 */
-	protected function create_contact( $email ) {
+	protected function create_contact( $email, $create_or_update = false ) {
 		$this->message = null;
 
 		$this->contact_data = array();
 		$properties = apply_filters( 'hubspot_contacts_properties', $this->get_properties_for_post(), 'create' );
-		$response = $this->send_post( 'contact', array( 'properties' => $properties ) );
+		if ( $create_or_update ) {
+			$email = false;
+			foreach ( $properties as $property ) {
+				if ( $property['property'] == 'email' ) {
+					$email = $property['value'];
```

Comment:
You're referencing `$property['value']` and `$property['property']` without checking their existence. Especially since `$properties` is filtered, you might want to do that. Up to you.